py/desitarget/targets.py

```diff
@@ -289,6 +289,92 @@ def decode_targetid(targetid):
 
     return objid, brickid, release, mock, sky
 
+############################################################
+def initial_priority_numobs(targets):
+    """highest initial priority and corresponding numobs for an array of target bits
```

Comment:
Just reading the code, this looks like it's doing the right thing, but I'll need to test-drive it and will likely have more input after the merge.